src/components/RequestGroupComposition/Configuration.vue

```diff
@@ -862,64 +923,15 @@ export default {
       return parameters;
     },
     generateDitherPattern: function() {
-      if (!_.isEmpty(this.errors)) {
-        alert('Please make sure your configuration is valid before generating a dither pattern for it');
-        return false;
-      }
-      this.dither.showDitherModal = true;
-      this.dither.status.loaded = false;
-      this.dither.status.error = false;
-      this.dither.status.notFound = false;
-      this.dither.status.errorResponse = {};
-      $.ajax({
-        method: 'POST',
-        url: `${this.observationPortalApiBaseUrl}/api/configuration/dither/`,
-        data: JSON.stringify(this.getDitherParameters(false)),
-        contentType: 'application/json'
-      })
-        .done(response => {
-          if (response.instrument_configs.length < 1) {
-            this.dither.status.notFound = true;
-          } else {
-            this.dither.expandedInstrumentConfigs = response.instrument_configs;
-          }
+      if (
+        this.checkReadyToGenerateExpansion('Please make sure your configuration is valid before generating a dither pattern for it', () => {
+          return !_.isEmpty(this.errors);
         })
-        .fail(response => {
-          this.dither.status.errorResponse = response.responseJSON;
-          this.dither.status.error = true;
-        })
-        .always(() => {
-          this.dither.status.loaded = true;
+      ) {
+        this.generateExpansion(`${this.observationPortalApiBaseUrl}/api/configurations/dither/`, this.getDitherParameters(false), response => {
+          return response.instrument_configs;
```

Comment:
We should just note that this only returns the instrument_configs, so it wouldn't get the obs_recipe if we decide to add that in the future for dithers since its in the configuration extra params...